src/r_things.c

```diff
@@ -941,33 +962,44 @@ void R_DrawSprite (vissprite_t* spr)
 
   //    for (ds=ds_p-1 ; ds >= drawsegs ; ds--)    old buggy code
 
-  for (ds=ds_p ; ds-- > drawsegs ; )  // new -- killough
-    {      // determine if the drawseg obscures the sprite
-      if (ds->x1 > spr->x2 || ds->x2 < spr->x1 ||
-          (!ds->silhouette && !ds->maskedtexturecol))
+  // [Woof!] e6y: optimization
+  if (drawsegs_xrange_size)
+  {
+    const drawseg_xrange_item_t *last = &drawsegs_xrange[drawsegs_xrange_count - 1];
+    drawseg_xrange_item_t *curr = &drawsegs_xrange[-1];
+    while (++curr <= last)
+    {
+      // determine if the drawseg obscures the sprite
+      if (curr->x1 > spr->x2 || curr->x2 < spr->x1)
         continue;      // does not cover sprite
 
-      r1 = ds->x1 < spr->x1 ? spr->x1 : ds->x1;
-      r2 = ds->x2 > spr->x2 ? spr->x2 : ds->x2;
+      ds = curr->user;
 
       if (ds->scale1 > ds->scale2)
-        {
-          lowscale = ds->scale2;
-          scale = ds->scale1;
-        }
+      {
+        lowscale = ds->scale2;
+        scale = ds->scale1;
+      }
       else
-        {
-          lowscale = ds->scale1;
-          scale = ds->scale2;
-        }
+      {
+        lowscale = ds->scale1;
+        scale = ds->scale2;
+      }
 
-      if (scale < spr->scale || (lowscale < spr->scale &&
-                    !R_PointOnSegSide (spr->gx, spr->gy, ds->curline)))
```

Comment:
Thanks for this, but please let's keep the original whitespace. This policy is actually very useful, especially for fixing desyncs.
Whitespace in MBF is not that bad, just Lee Killough has 8-space tabs enabled.

Reply:
Done. Is there something else for now?

Reply:
Thank you! I haven't tested it yet, I'll try it in a couple of hours 👍